lib/ocpp/v201/charge_point.cpp

```diff
@@ -220,11 +220,13 @@ void ChargePoint::on_firmware_update_status_notification(int32_t request_id,
     }
 
     if (this->firmware_status_before_installing == req.status) {
-        this->firmware_status = FirmwareStatusEnum::InstallScheduled;
-        req.status = firmware_status;
-        ocpp::Call<FirmwareStatusNotificationRequest> call(req, this->message_queue->createMessageId());
-        this->send_async<FirmwareStatusNotificationRequest>(call);
-
+        const auto transaction_active = this->any_transaction_active(std::nullopt);
+        if (transaction_active) {
+            this->firmware_status = FirmwareStatusEnum::InstallScheduled;
```

Comment:
I will test your solution. But as we are already sending InstallScheduled from our 'core', will that go well?

Anyway, will test, let you know!

Reply:
That might be an issue, maybe we can use another method of determining if transactions are still running that doesn't involve this part of the code in libocpp

Reply:
This is indeed failing L_15. Because InstallScheduled is already sent (from the 'core' application), it will send it again.

It is a bit strange that all firmware statusses are sent from 'core' and only one is sent from here isn't it?
But keeping a variable that stores the last sent status will probably do the trick as well.

Reply:
You're right, this isn't the proper place for it. I'll remove the commit from this PR and figure something out for L_15